sdk/core/azure-core-test/src/main/java/com/azure/core/test/http/PlaybackClient.java

```diff
@@ -49,7 +51,14 @@ public PlaybackClient(RecordedData recordedData, Map<String, String> textReplace
         Objects.requireNonNull(recordedData, "'recordedData' cannot be null.");
 
         this.recordedData = recordedData;
-        this.textReplacementRules = textReplacementRules == null ? new HashMap<>() : textReplacementRules;
+        this.textReplacementRules = new HashMap<>();
+        if (textReplacementRules != null) {
+            // Compile the replacement rules into Patterns as they'll be used as String.replaceAll functionality which
+            // compiles the target pattern anyways.
+            for (Map.Entry<String, String> kvp : textReplacementRules.entrySet()) {
+                this.textReplacementRules.put(Pattern.compile(kvp.getKey()), kvp.getValue());
```

Comment:
is possible to have Pattern.compile(null)?

Reply:
Nope, and this would have failed previously when using the `String.replaceAll`.